### PR TITLE
Fix: Remove rules for NestJS applications

### PR DESCRIPTION
**Pull request by ccoeurderoy**
### Description

In order to be easily compatible with NestJS applications, we had to remove some rules:

- `no-unbound-method`: removed for static methods

_Example_:

```ts
  /**
   * Handle Pubsub error
   * @param error Error thrown
   */
  private static onError(error: Error): void {
    throw new InternalServerErrorException(`An error occurred on PubSub: ${error.message}`);
  }


  public method(...) {
      ...
      this.pubSubService.emit(formattedMessage).catch(FolderPubSubService.onError);
  }
```

It raises an error 

- `no-unnecessary-class`: lots of files are declared like:

```ts
/**
 * Define the app module
 */
@Module({ ... })
export class AppModule {}
```

- `no-parameter-property`: we should be able to instantiate class properties like:

```ts
@Injectable()
export class GCSService implements OnModuleInit {
  constructor(private readonly config: ConfigService, @Inject('winston') private readonly logger: Logger) {}
```

- `object-literal-sort-keys`: model properties could be defined in a specific order, such as mongo queries

### CODE REVIEW

File: tslint-config.json [hunk 1]
```diff
@@ -83,9 +83,9 @@
     "no-submodule-imports": false,
     "no-switch-case-fall-through": true,
     "no-this-assignment": true,
-    "no-unbound-method": true,
-    "no-unnecessary-class": true,
-    "no-unsafe-any": true,
+    "no-unbound-method": [true, "ignore-static"],
+    "no-unnecessary-class": [true, "allow-empty-class", "allow-empty-constructor"],
+    "no-unsafe-any": false,
```
Comment (na-ji) [inline tslint-config.json]: Too bad we have to remove this rule. It's an important one.

**Review by na-ji — APPROVED** [review]

File: tslint-config.json (hunk 1 above)
Comment (ccoeurderoy) [reply tslint-config.json]: I agree! But if an external module uses the any type, this rule always raises an error 😕

Comment (na-ji) [reply tslint-config.json]: I think I got around this rule by casting the parameters, but I did not have to do it a lot of times.

Comment (ccoeurderoy) [reply tslint-config.json]: Well, with decorators, it is pretty annoying. For instance:

```ts
@ApiOkResponse({
    description: 'Successful operation',
  })
  @ApiBadRequestResponse({
    description: 'Bad request',
  })
  @ApiNotFoundResponse({
    description: 'Not found',
  })
  public async find(@Param('folderId', ObjectIDValidationPipe) folderId: string): Promise<Document[]> {
    return this.documentsService.find(folderId);
  }
```

Each decorator returns any type, how can we cast properly these return types?

Comment (na-ji) [reply tslint-config.json]: We can't cast decorators. The only solution I found is to overwrite the typings of the modules. 😢

Comment (pierreneveu) [reply tslint-config.json]: Is it possible to override in each project this field instead of setting it to `false` for all of them? I think it's better

Comment (ccoeurderoy) [reply tslint-config.json]: Yes indeed, we can do this!

**Review by na-ji — APPROVED** [review]

**Review by pierreneveu — APPROVED** [review]

**Review by diephil — APPROVED** [review]

**Pull request merged**